src/pilot/templates/post-install.yaml

```diff
@@ -1,60 +1,60 @@
-# Copyright (c) 2017 Dell Inc. or its subsidiaries.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-heat_template_version: 2014-10-16
-
-description: >
-  Example extra config for post-deployment
-
-parameters:
-  servers:
-    type: json
-
-resources:
-  ExtraDeployments:
-    type: OS::Heat::StructuredDeployments
-    properties:
-      servers:  {get_param: servers}
-      config: {get_resource: ExtraConfig}
-      # Do this on CREATE/UPDATE (which is actually the default)
-      actions: ['CREATE', 'UPDATE']
-
-  ExtraConfig:
-    type: OS::Heat::SoftwareConfig
-    properties:
-      group: script
-      config: |
-        #!/bin/bash
-
-        set -x
-        if [[ $HOSTNAME =~ "compute" ]] ; then
-          VCPUPINSET=$(crudini --get /root/parameters.ini DEFAULT novacpus)
-          crudini --set /etc/nova/nova.conf DEFAULT vcpu_pin_set $VCPUPINSET
-          systemctl restart openstack-nova-compute.service
-          tuned_service=/usr/lib/systemd/system/tuned.service
-          grep -q "network.target" $tuned_service
-          if [ "$?" -eq 0 ]; then
-            sed -i '/After=.*/s/network.target//g' $tuned_service
-          fi
-          grep -q "Before=.*network.target" $tuned_service
-          if [ ! "$?" -eq 0 ]; then
-            grep -q "Before=.*" $tuned_service
-            if [ "$?" -eq 0 ]; then
-              sed -i 's/^\(Before=.*\)/\1 network.target openvswitch.service/g' $tuned_service
-            else
-              sed -i '/After/i Before=network.target openvswitch.service' $tuned_service
-            fi
-          fi
-          systemctl daemon-reload
-        fi
+# Copyright (c) 2016-2018 Dell Inc. or its subsidiaries.
```

Comment:
Why do we need a post-install script? What is its purpose?

Reply:
To apply vcpus (after subtracting pmd cpus)  in nova.conf 
To add network.target and systemd-sysctl.service to tuned.service
This is related to tuned-adm

Reply:
see my comment below. Why do we need to set up these services after the deployment? This is part of the performance enhancement?

Reply:
Note that this is not a part of performance optimization. 
This is related to tuned-adm cpu-partitioning profile which is required to set for dpdk.  
I think xlfow guys can explain this in details.